modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5544,7 +5544,7 @@ Error GDScriptAnalyzer::analyze() {
 	// Apply annotations.
 	for (GDScriptParser::AnnotationNode *&E : parser->head->annotations) {
 		resolve_annotation(E);
-		E->apply(parser, parser->head);
+		E->apply(parser, parser->head, nullptr);
```

Comment:
`parser->head` is a `ClassNode *`, so this line should be:

```cpp
E->apply(parser, parser->head, nullptr);
```

Reply:
No, `p_class` is the class that `p_target` is in. The root class is **not** in any other class. This is a subtle detail, but for example if an annotation is applied to an inner class (we don't have such a case right now), then `p_target` will be the class, and `p_class` will be its outer class. Therefore, if you do not need the _context_ class, then it is important to use `p_target` rather than `p_class`, even this requires `static_cast`.

Reply:
Thanks, I missed that detail before. I've updated my PR and add a comment in the abstract annotation apply method for why we are using p_target instead of p_class.